### PR TITLE
Adding Radio Buttons

### PR DESCRIPTION
**Pull request by ThomasCanning**
Adding Radio Buttons. Create a group by setting the "name" parameter to the same thing. The radio buttons are referenced by their "value" and labeled with "label". The initailaly set button can also be changed. Set parameters is first and is last to add rounding to the corners of the first and last elements.

### CODE REVIEW

**Comment by WilfSilver** [comment]
@ThomasCanning sammy told me this was a draft so have marked it as such. Do you agree with this decision/do you think its ready for a review?

**Comment by ThomasCanning** [comment]
I think it's good to go, it's customised the same as in the figma pretty
much

File: packages/frontend/src/components/radio_button.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// RadioButton.tsx
```
Comment (WilfSilver) [inline packages/frontend/src/components/radio_button.tsx:1]: This can be removed

File: packages/frontend/src/components/radio_button.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import type { ReactNode } from "react";
+
+import React from "react";
```
Comment (WilfSilver) [inline packages/frontend/src/components/radio_button.tsx:4]: `ReactNode` is not used, see [the failing check](https://github.com/ESP-Number-One/MatchyoSports/actions/runs/7976717417/job/21777924312#step:3:86). However it seems you use `React.ReactNode`, so you can instead remove the `React` import and remove replace `React.ReactNode` with `ReactNode` which I find more readable.

Note you will have to import `ChangeEvent`.

File: packages/frontend/src/components/radio_button.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  isFirst?: boolean;
+  isLast?: boolean;
+}
+
+export function RadioButton({
+  name,
+  value,
+  label,
+  icon,
+  checked = false,
```
Comment (WilfSilver) [inline packages/frontend/src/components/radio_button.tsx:22]: I feel like this should be stored as a state inside the this object itself.

**Review by WilfSilver — CHANGES_REQUESTED** [review]
I worry slightly about the implementation as its a bit more complicated to setup than I was expecting (as we have to wrap the buttons in a `div` with class `grid`)

**Comment by WilfSilver** [comment]
I have rebased your branch, on main.

Please run `git pull --force` before making any other changes

**Comment by WilfSilver** [comment]
This was merged with #52

**Pull request closed without merging**